src/test/java/com/verily/genomewarp/utils/GvcfToVcfAndBedTest.java

```diff
@@ -34,65 +35,81 @@
   private static final String SAMPLE_GVCF = "utils/smallNA12889_S1.genome_g.vcf";
   private static final String EXPECTED_VCF = "utils/smallNA12889_S1.genome.vcf";
   private static final String EXPECTED_BED = "utils/smallNA12889_S1.genome.bed";
+  private static final String MALFORMED_GVCF = "utils/malformed_g.vcf";
 
   @Rule
   public TemporaryFolder tempFolder = new TemporaryFolder();
 
   @Test
-  public void testSaveVcfAndBedFromGvcf() throws Exception {
+  public void testSaveVcfAndBedFromGvcf() throws IOException {
     File sampleGvcf =
         new File(GvcfToVcfAndBedTest.class.getClassLoader().getResource(SAMPLE_GVCF).getFile());
-    String rawQueryVcf =
-        // GvcfToVcfAndBedTest.class.getClassLoader().getResource(SAMPLE_GVCF).getFile();
-        sampleGvcf.getCanonicalPath();
+    String rawQueryGvcf = sampleGvcf.getCanonicalPath();
     String tempFolderPath = tempFolder.getRoot().getCanonicalPath();
     String outVcf = tempFolderPath + File.separator + "from_gvcf.vcf";
     String outBed = tempFolderPath + File.separator + "from_gvcf.bed";
-    boolean isGvcf = GvcfToVcfAndBed.saveVcfAndBedFromGvcf(rawQueryVcf, outVcf, outBed);
-    assertTrue(isGvcf);
+    assertTrue(GvcfToVcfAndBed.saveVcfAndBedFromGvcf(rawQueryGvcf, outVcf, outBed));
+
     // compare files line by line
     String line1, line2;
 
     // compare VCF files
     File expVcf =
         new File(GvcfToVcfAndBedTest.class.getClassLoader().getResource(EXPECTED_VCF).getFile());
-    BufferedReader expVcfBR = Files.newBufferedReader(Paths.get(expVcf.getCanonicalPath()), UTF_8);
-    BufferedReader gotVcfBR = Files.newBufferedReader(Paths.get(outVcf), UTF_8);
-    while ((line1 = gotVcfBR.readLine()) != null) {
-      line2 = expVcfBR.readLine();
-      assertTrue(line2 != null);
-      assertEquals(line1, line2);
+    try (
+        BufferedReader expVcfBR =
+            Files.newBufferedReader(Paths.get(expVcf.getCanonicalPath()), UTF_8);
+        BufferedReader gotVcfBR = Files.newBufferedReader(Paths.get(outVcf), UTF_8)) {
+      while ((line1 = gotVcfBR.readLine()) != null) {
```

Comment:
Optional: You could extract the logic of lines 63-68 into a private helper function assertBufferedReadersEqual(BufferedReader r1, BufferedReader r2) since the identical logic is used for the VCF and BED files.